src/pages/home/report/PureReportActionItem.tsx

```diff
@@ -0,0 +1,1113 @@
+    useEffect(() => {
+        if (index !== 0 || !isActionableWhisper) {
+            return;
+        }
+
+        if (prevActionResolution !== (hasResolutionInOriginalMessage ? originalMessage.resolution : null)) {
+            reportScrollManager.scrollToIndex(index);
+        }
+    }, [index, originalMessage, prevActionResolution, reportScrollManager, isActionableWhisper, hasResolutionInOriginalMessage]);
+
+    const toggleReaction = useCallback(
+        (emoji: Emoji, ignoreSkinToneOnCompare?: boolean) => {
+            Report.toggleEmojiReaction(reportID, action, emoji, emojiReactions, undefined, ignoreSkinToneOnCompare);
+        },
+        [reportID, action, emojiReactions],
+    );
+
+    const contextValue = useMemo(
+        () => ({
+            anchor: popoverAnchorRef.current,
+            report: {...report, reportID: report?.reportID ?? ''},
+            reportNameValuePairs,
+            action,
+            transactionThreadReport,
+            checkIfContextMenuActive: toggleContextMenuFromActiveReportAction,
+            isDisabled: false,
+        }),
+        [report, action, toggleContextMenuFromActiveReportAction, transactionThreadReport, reportNameValuePairs],
+    );
+
+    const attachmentContextValue = useMemo(() => ({reportID, type: CONST.ATTACHMENT_TYPE.REPORT}), [reportID]);
+
+    const mentionReportContextValue = useMemo(() => ({currentReportID: report?.reportID ?? '-1'}), [report?.reportID]);
+
+    const actionableItemButtons: ActionableItem[] = useMemo(() => {
+        if (ReportActionsUtils.isActionableAddPaymentCard(action) && !doesUserHavePaymentCardAdded() && shouldRenderAddPaymentCard()) {
+            return [
+                {
+                    text: 'subscription.cardSection.addCardButton',
+                    key: `${action.reportActionID}-actionableAddPaymentCard-submit`,
+                    onPress: () => {
+                        Navigation.navigate(ROUTES.SETTINGS_SUBSCRIPTION_ADD_PAYMENT_CARD);
+                    },
+                    isMediumSized: true,
+                    isPrimary: true,
+                },
+            ];
+        }
+
+        if (!isActionableWhisper && (!ReportActionsUtils.isActionableJoinRequest(action) || ReportActionsUtils.getOriginalMessage(action)?.choice !== ('' as JoinWorkspaceResolution))) {
+            return [];
+        }
+
+        if (ReportActionsUtils.isActionableTrackExpense(action)) {
+            const transactionID = ReportActionsUtils.getOriginalMessage(action)?.transactionID;
+            return [
+                {
+                    text: 'actionableMentionTrackExpense.submit',
+                    key: `${action.reportActionID}-actionableMentionTrackExpense-submit`,
+                    onPress: () => {
+                        ReportUtils.createDraftTransactionAndNavigateToParticipantSelector(transactionID ?? '0', reportID, CONST.IOU.ACTION.SUBMIT, action.reportActionID);
```

Comment:
Same here and for the other calls to `createDraftTransactionAndNavigateToParticipantSelector` below

Reply:
done